libpf/pfelf/file.go

```diff
@@ -732,6 +729,17 @@ func (ph *Prog) DataReader(maxSize uint) (io.Reader, error) {
 	return bytes.NewReader(p), nil
 }
 
+// ReadAt implements the io.ReaderAt interface
+func (sh *Section) ReadAt(p []byte, off int64) (n int, err error) {
+	if off < 0 || uint64(off) >= sh.FileSize {
+		return 0, io.EOF
+	}
+	if uint64(off)+uint64(len(p)) > sh.FileSize {
+		p = p[:sh.FileSize-uint64(off)]
```

Comment:
Should we explicitly return `EOF` here? The following `ReadAt` will not for this case (as it doesn't know anything about `sh.FileSize`).

Reply:
Yes. Per `io.ReaderAt` contract an error needs to be returned if read is short. I've added this.